### PR TITLE
Initial support for TKC Alice

### PR DESCRIPTION
**Pull request by TerryMathews**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tkc/alice/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# TKC Alice
+
+![Alice](https://i.imgur.com/U2x7TQ8.jpg)
+
+The TKC Alice is an ATMega32u4 powered, USB Type C, Alice compatible PCB and was created specifically for the OSA case, but with considerations to make it compatible with the Lubrigante acrylic cases (except for the Type C connector it should also be compatible with original TGR Alice cases).
+
+Keyboard Maintainer: [Terry Mathews](https://github.com/TerryMathews/)  
+Hardware Supported: TKC Alice PCB  
+Hardware Availability: https://thekey.company/collections/alice-keyboard
```
Comment (fauxpark) [inline keyboards/tkc/alice/README.md:9]:
```suggestion
* Keyboard Maintainer: [Terry Mathews](https://github.com/TerryMathews/)
* Hardware Supported: TKC Alice PCB
* Hardware Availability: https://thekey.company/collections/alice-keyboard
```

File: keyboards/tkc/alice/alice.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+  // put your per-action keyboard code here
+  // runs for every action, just before processing by the firmware
+
+  return process_record_user(keycode, record);
+}
+*/
+
```
Comment (fauxpark) [inline keyboards/tkc/alice/alice.c:46]:
```suggestion

```

File: keyboards/tkc/alice/alice.c [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+void matrix_init_board(void){
```
Comment (fauxpark) [inline keyboards/tkc/alice/alice.c:47]:
```suggestion
void matrix_init_board(void) {
```

File: keyboards/tkc/alice/alice.c [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        writePinLow(C6);
+    }
+    if (IS_LED_ON(usb_led, USB_LED_SCROLL_LOCK)) {
+        writePinHigh(B6);
+    } else {
+        writePinLow(B6);
+    }
+  led_set_user(usb_led);
+}
+
```
Comment (fauxpark) [inline keyboards/tkc/alice/alice.c:73]:
```suggestion
bool led_update_kb(led_t led_state) {
    if (led_update_user(led_state)) {
        writePin(C7, led_state.num_lock);
        writePin(C6, led_state.caps_lock);
        writePin(B6, led_state.scroll_lock);
    }
    return true;
}
```

File: keyboards/tkc/alice/config.h [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+#define LCD_DATA3_PIN    7            //< pin for 4bit data bit 3
+#define LCD_RS_PORT      LCD_PORT     //< port for RS line
+#define LCD_RS_PIN       3            //< pin  for RS line
+#define LCD_RW_PORT      LCD_PORT     //< port for RW line
+#define LCD_RW_PIN       2            //< pin  for RW line
+#define LCD_E_PORT       LCD_PORT     //< port for Enable line
+#define LCD_E_PIN        1            //< pin  for Enable line
+#endif
+*/
+
```
Comment (fauxpark) [inline keyboards/tkc/alice/config.h:248]:
```suggestion
```

File: keyboards/tkc/alice/info.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+    "keyboard_name": "TKC Alice",
+    "url": "",
+    "maintainer": "qmk",
+    "width": 18.25,
+    "height": 5,
+    "layouts": {
+    "LAYOUT_default_ansi": {
+            "key_count": 64,
```
Comment (fauxpark) [inline keyboards/tkc/alice/info.json:9]:
```suggestion
```

File: keyboards/tkc/alice/info.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        },
+        "LAYOUT_split_backspace": {
+            "key_count": 65,
```
Comment (fauxpark) [inline keyboards/tkc/alice/info.json:13]:
```suggestion
```

File: keyboards/tkc/alice/info.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        },
+        "LAYOUT_split_right_shift": {
+            "key_count": 65,
```
Comment (fauxpark) [inline keyboards/tkc/alice/info.json:17]:
```suggestion
```

File: keyboards/tkc/alice/info.json [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        },
+        "LAYOUT_all": {
+            "key_count": 66,
```
Comment (fauxpark) [inline keyboards/tkc/alice/info.json:21]:
```suggestion
```

File: keyboards/tkc/alice/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+           KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS,                   KC_TRNS,                   KC_TRNS,                            KC_TRNS
+  ),
+
+[3] = LAYOUT_all(
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+           KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+           KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS,                   KC_TRNS,                   KC_TRNS,                            KC_TRNS
+  ),
```
Comment (fauxpark) [inline keyboards/tkc/alice/keymaps/default/keymap.c:48]:
```suggestion
```

File: keyboards/tkc/alice/keymaps/via/rules.mk [hunk 11]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+MOUSEKEY_ENABLE = no
```
Comment (fauxpark) [inline keyboards/tkc/alice/keymaps/via/rules.mk:2]:
```suggestion
```

File: keyboards/tkc/alice/rules.mk [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs
+LTO_ENABLE = yes			# Reduces compile size
```
Comment (fauxpark) [inline keyboards/tkc/alice/rules.mk:34]:
```suggestion
BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
LTO_ENABLE = yes            # Reduces compile size
```

File: keyboards/tkc/alice/alice.c (hunk 4 above)
Comment (TerryMathews) [reply keyboards/tkc/alice/alice.c]: @fauxpark Can't apply suggestion - says outdated.

File: keyboards/tkc/alice/alice.c [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "alice.h"
+void matrix_init_board(void) {
```
Comment (fauxpark) [inline keyboards/tkc/alice/alice.c:17]:
```suggestion

void keyboard_pre_init_kb(void) {
```

File: keyboards/tkc/alice/alice.c [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+    } else {
+        writePinLow(C6);
+    }
+    if (IS_LED_ON(usb_led, USB_LED_SCROLL_LOCK)) {
+        writePinHigh(B6);
+    } else {
+        writePinLow(B6);
+    }
+  led_set_user(usb_led);
+}
```
Comment (fauxpark) [inline keyboards/tkc/alice/alice.c:43]:
```suggestion
bool led_update_kb(led_t led_state) {
    if (led_update_user(led_state)) {
        writePin(C7, led_state.num_lock);
        writePin(C6, led_state.caps_lock);
        writePin(B6, led_state.scroll_lock);
    }
    return true;
}
```

File: keyboards/tkc/alice/alice.c [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "alice.h"
+
+void keyboard_pre_init_kb(void) {
+    setPinOutput(C7);
+    setPinOutput(C6);
+    setPinOutput(B6);
+}
```
Comment (Erovia) [inline keyboards/tkc/alice/alice.c:22]:
```suggestion

    keyboard_pre_init_user();
}
```

**Comment by TerryMathews** [comment]
I need to put this PR on hold for the time being.

**Comment by TerryMathews** [comment]
OK this is ready.

File: keyboards/tkc/osav2/info.json [hunk 16]
```diff
@@ -0,0 +1,23 @@
+{
+    "keyboard_name": "TKC OSA v2",
+    "url": "",
+    "maintainer": "qmk",
+    "width": 18.25,
+    "height": 5,
+    "layouts": {
+    "LAYOUT_default_ansi": {
+            "key_count": 64,
```
Comment (fauxpark) [inline keyboards/tkc/osav2/info.json:9]:
```suggestion
```

File: keyboards/tkc/osav2/info.json [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+        },
+        "LAYOUT_split_backspace": {
+            "key_count": 65,
```
Comment (fauxpark) [inline keyboards/tkc/osav2/info.json:13]:
```suggestion
```

**Pull request merged**